### PR TITLE
Added Kitchen tests and Travis

### PR DESCRIPTION
**Pull request by Martin819**

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: apache/server/service/init.sls [hunk 1]
```diff
@@ -57,6 +59,14 @@ apache_service:
   - require:
     - pkg: apache_packages
 
+{% else %}
+
+apache_service:
+  service.dead:
```
Comment (fpytloun) [inline apache/server/service/init.sls]: noservices means - do not touch services, so it shouldn't even ensure they are dead (eg. they may be running but you just want to skip it's restart and want to do that manually, etc.)

File: apache/map.jinja [hunk 2]
```diff
@@ -57,7 +57,44 @@
...
+        'service_user': 'www-data',
+        'service_group': 'www-data',
+    },
+    'xenial': {
+        'pkgs': ['apache2'],
+        'service': 'apache2',
+        'mod_wsgi': 'libapache2-mod-wsgi',
+        'mod_php': 'libapache2-mod-php7.0',
+        'module_php': 'php7.0',
+    'mod_perl': 'libapache2-mod-perl2',
```
Comment (fpytloun) [inline apache/map.jinja]: wrong indent

File: apache/map.jinja [hunk 3]
```diff
@@ -57,7 +57,44 @@
...
     },
-}, merge=salt['pillar.get']('apache:server')) %}
+}, merge=salt['grains.filter_by']({
+    'trusty': {
+        'pkgs': ['apache2', 'apache2-utils'],
+        'service': 'apache2',
+        'mod_wsgi': 'libapache2-mod-wsgi',
+        'mod_php': 'libapache2-mod-php5',
+        'module_php': 'php5',
+    'mod_perl': 'libapache2-mod-perl2',
```
Comment (fpytloun) [inline apache/map.jinja]: wrong indent

**Pull request merged**